File: src/commands/inline/inline.command.ts
import { queryResults } from './query-results';
import { Command } from '../command';
import * as ru from 'convert-layout/ru';
import { Bot } from 'grammy';

export class InlineCommand extends Command {
  constructor(bot: Bot) {
    super(bot);
  }

  init() {
    this.bot.on('inline_query', (ctx) => {
      let query = ctx.inlineQuery.query;

      if (/^[a-z]+$/ui.test(query)) {
        query = ru.fromEn(query);
      }

      const filtered = queryResults.filter(v => {
        return (new RegExp(query, 'gium').test(v.title));
      });

      return ctx.answerInlineQuery(filtered);
    });
  }
}

File: bin/commands/inline/inline_command.dart
import 'package:get_it/get_it.dart';
import 'package:teledart/model.dart';

import '../../services/telegram.service.dart';
import 'query_results.dart';

class InlineCommand {
  InlineCommand() {
    final bot = GetIt.I.get<TelegramService>();

    bot.teledart.onInlineQuery().listen((inlineQuery) =>
        inlineQuery.answer(InlineCommand.getResults(inlineQuery.query)));
  }

  static List<InlineQueryResultArticle> getResults(String query) {
    final queryRegExp = RegExp(query, caseSensitive: false, unicode: true);

    return queryResults
        .where((element) => queryRegExp.hasMatch(element.title))
        .toList();
  }
}
